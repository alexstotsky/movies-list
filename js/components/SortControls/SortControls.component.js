import React from 'react'
import {
  View, TouchableOpacity, StyleSheet,
} from 'react-native'
import Icon from 'react-native-vector-icons/Octicons'
import PropTypes from 'prop-types'

import { styleSheet } from '../../utils/commonPropTypes'

import SubTitle from '../Subtitle'

class SortControls extends React.Component {
  static propTypes = {
    containerStyle: styleSheet,
    direction: PropTypes.oneOf(['row', 'column']).isRequired,
    controls: PropTypes.arrayOf(PropTypes.shape({
      iconUp: PropTypes.shape({
        group: PropTypes.string,
        name: PropTypes.string,
      }),
      iconDown: PropTypes.shape({
        group: PropTypes.string,
        name: PropTypes.string,
      }),
      asc: PropTypes.bool,
      active: PropTypes.bool,
      label: PropTypes.string,
      controlStyle: styleSheet,
      onPress: PropTypes.func.isRequired,
    })).isRequired,
  }

  static defaultProps = {
    direction: 'row',
    controls: [],
  }

  render() {
    const { direction, containerStyle, controls } = this.props
    return (
      <View style={[styles[direction], containerStyle]}>
        {
          controls.map((control, index) => {
            return (
              <View key={index} style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
                <TouchableOpacity style={styles.row} onPress={control.onPress}>
                  <SubTitle subtitle={control.label} />
                  {
                    control.active
                      ? (control.asc
                        ? <Icon size={15} name={control.iconUp.name} color='black' />
                        : <Icon size={15} name={control.iconDown.name} color='black' />
                      )
                      : <Icon size={15} name={control.iconUp.name} color='gray' />
                  }
                </TouchableOpacity>
              </View>
            )
          })
        }
      </View>
    )
  }
}

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
  },
  column: {
    flexDirection: 'column',
  },
})

export default SortControls
