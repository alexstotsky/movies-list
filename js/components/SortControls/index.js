import SortControls from './SortControls.component'

export default SortControls
