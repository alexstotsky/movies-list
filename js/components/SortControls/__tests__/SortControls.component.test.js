import React from 'react'
import { shallow } from 'enzyme'

import SortControls from '../SortControls.component'

describe('Testing SortControls component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <SortControls />,
    )
    expect(wrapper).toMatchSnapshot()
    wrapper.setProps({
      containerStyle: {
        width: '100%',
      },
      direction: 'column',
      controls: [
        {
          iconUp: {
            name: 'triangle-up',
          },
          iconDown: {
            name: 'triangle-up',
          },
          asc: true,
          active: true,
          label: 'asd',
          onPress: () => {},
          containerStyle: {
            width: '100%',
          },
          labelStyle: {
            width: '100%',
          },
        },
      ],
    })
    expect(wrapper).toMatchSnapshot()
  })
})
