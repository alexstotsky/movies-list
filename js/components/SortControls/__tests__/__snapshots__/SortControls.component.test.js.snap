// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SortControls component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
/>
`;

exports[`Testing SortControls component renders correctly 2`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "column",
      },
      Object {
        "width": "100%",
      },
    ]
  }
>
  <View
    key="0"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <SubTitle
        containerStyle={
          Object {
            "width": "100%",
          }
        }
        subtitle="asd"
        subtitleStyle={
          Object {
            "width": "100%",
          }
        }
      />
      <Icon
        allowFontScaling={false}
        color="black"
        name="triangle-up"
        size={15}
      />
    </TouchableOpacity>
  </View>
</View>
`;
